rfcs/text/0009_screenshot_mode_service.md

```diff
@@ -0,0 +1,244 @@
+user using the page interactively, and when it's not. Sometimes the page is
+loaded for display-purposes: there is no user in control of the
+browser and no need for interactivity. Having Kibana support that purpose
+allows higher-level features to be built on top of rapidly-available images of
+Kibana pages. This turns out to be crucial for features like scheduled PDF
+report generation, and automated alerts with embedded images.
+
+As a bridge between the Kibana applications screenshot-capture features is
+needed as a core service in Kibana. This RFC names that service the Screenshot
+Mode Service (name TBD). Applications would read from this service while
+loading UI components into the browser, and help the application make
+case-by-case bases for loading components and the parameters that get passed
+into the components.
+
+In most cases, the information coming from this service would help the
+application which UI components should **not** be loaded, as applications will
+use the service response for skipping components that aren't needed for
+presenting the data in a screenshot, and thus work towards better performance
+for the display-only purpose.
+
+# Basic example
+
+When Kibana loads initially, there is a Newsfeed component in the UI that
+checks internally cached records to see if it must fetch the Elastic News
+Service for newer items. When the Screenshot Mode Service is implemented, the
+Newsfeed component has a source of information to check on whether or not it
+should load in the Kibana UI. If it can avoid loading, it avoids an unecessary
+HTTP round trip, which weigh heavily on performance.
+
+# Motivation
+
+Kibana Reporting is a commercial feature in Elastic and is highly capable of
+loading the application pages and converting them into a screenshot for PNG or
+PDF export. However, the way it works has downsides with performance,
+maintainability, and expanding it as a tool that can power higher-level
+features. The solution to those downsides is to have Kibana pages themselves
+become more capable at presenting themselves for screenshot capture reports.
+With the Screenshot Mode Service available in core, Reporting could drop the
+task that it currently has which hurt performance: wasted rendering that is
+replaced with custom styles that make the page "reportable."
+
+The technical advantage of having such as service also leads to making Kibana
+application pages "printable", in the sense that sending the page to a printer
+for a hard copy results in something more meaningful and specialized for paper
+than today's Kibana can guarantee. This isn't a big concern for Kibana since
+there isn't the expectation to improve printing Kibana, but that technical
+direction is appropriate for improving PDF report generation.
+
+# Detailed design
+
+The Screenshot Mode Service is a callable API availble in the context of a
+request on the server side, and as a pubic coreSetup API Javascript.
```

Comment:
I think `and as a pubic coreSetup API Javascript` is `and as a pubic coreSetup JavaScript API` ?